src/erlcloud_aws.erl

```diff
@@ -1572,3 +1574,50 @@ maybe_imdsv2_session_token(Config) ->
         {ok, Token} -> Token;
         _Error -> undefined
     end.
+
+%%%%%%%%%%%%%%%%%
+
+% Takes the query response and turns it into a map if desired
+parse_response({ok, Response}, map) ->
+    {ok, _Res} = erlcloud_xml:xml_to_map(Response);
+parse_response({ok, Response}, _) ->
+    {ok, Response};
+parse_response({error, _} = ErrRes, _Format) -> 
+    ErrRes.
+
+concat_key(<<>>, Key) when is_bitstring(Key); is_atom(Key)  ->
+    to_bitstring(Key);
+concat_key(<<>>, Key) when is_list(Key) -> 
+    Key;
+concat_key(ParentKey, Key) ->
+    BiParentKey = to_bitstring(ParentKey),
+    BiKey = to_bitstring(Key),
+    <<BiParentKey/binary, ".", BiKey/binary>>.
+
+%%% Conversions
+to_bitstring(In) when is_bitstring(In) ->
+    In;
+to_bitstring(In) when is_list(In) ->
+    list_to_binary(In);
+to_bitstring(In) when is_atom(In) ->
+    erlang:atom_to_binary(In, utf8).
+
+process_params(Params) ->
+    process_params(Params, <<>>, <<>>).
+process_params(Params, ParentKey) ->
+    process_params(Params, ParentKey, <<>>).
+process_params(Params, ParentKey, ListPrependStr) when is_map(Params) ->
+    process_params2(maps:to_list(Params), ParentKey, ListPrependStr);
+    
+process_params(Params, ParentKey, ListPrependStr) when is_list(Params) ->
+    Indexes = lists:seq(1, length(Params)),
+    BinIndexes = [ erlcloud_aws:concat_key(ListPrependStr, integer_to_binary(Index)) || Index <- Indexes],
+    process_params2(lists:zip(BinIndexes, Params), ParentKey, ListPrependStr);
```

Comment:
What's the thought for  ParentKey and/or ListPrependStr. Am I missing somewhere where one/both are defined or definable?

Reply:
LIstPrependStr is an optional argument you may need to include for some of Amazons API's as they handle the unfolding of lists in different ways - an example of this is for the autoscale service: https://github.com/jcsawyer123/erlcloud/blob/7a392c4260717bdb5e08069ed296f92bbc2cfb57/src/erlcloud_as.erl#L814

ParentKey may be defined upfront to ensure all elements are defined as "$ParentKey.myKey", otherwise it is used recursively when flattening nested map structures. 

An example of its usage can be found in the test: https://github.com/jcsawyer123/erlcloud/blob/7a392c4260717bdb5e08069ed296f92bbc2cfb57/test/erlcloud_aws_tests.erl#L998